contents/tutorials/nuxt-feature-flags.md

```diff
@@ -0,0 +1,247 @@
+
+In our example, we fetch our feature flag and update the text on the page based on its value:
+
+```vue file=app.vue
+<template>
+  <main>
+    <h1>{{ title }}</h1>
+  </main>
+</template>
+
+<script setup>
+
+const title = ref('No variant');
+onMounted(() => {
+  const { $posthog } = useNuxtApp()
+  if ($posthog) {
+    const posthog = $posthog()
+    if (posthog.isFeatureEnabled('my-cool-flag')) {
+      title.value = 'Our flag is enabled!';
+    } else {
+      title.value = 'Our flag is disabled!';
+    }
+  }
+});
+
+</script>
+```
+
+When you run your app now, you should see the updated text.
+
+![New app after adding the flag](../images/tutorials/nuxt-feature-flags/flag-enabled-in-app.png)
+
+## Implementing flag code on the server side
+
+You may notice the text flickers when you load your app. This is because PostHog is making a request on the client side to fetch the flag value. To prevent this, there are two alternatives:
+
+1. Fetch the flag on the server side.
+2. [Bootstrap](/docs/feature-flags/bootstrapping) PostHog on the client side (not covered in this tutorial).
+
+We'll show you how to fetch the flag on the server side. First, install the [`posthog-node` SDK](/docs/libraries/node):
+
+```bash
+npm install posthog-node
+```
+
+Then, we use the PostHog Node library to fetch the feature flag using [`useAsyncData`](https://nuxt.com/docs/api/composables/use-async-data). Replace the code in `app.vue` with the following:
+
+```vue file=app.vue
+<template>
+  <main>
+    <h1>{{ title }}</h1>
+  </main>
+</template>
+
+<script setup>
+import { useAsyncData, useRuntimeConfig } from 'nuxt/app';
+import { PostHog } from 'posthog-node';
+
+const { data: titleData, error } = await useAsyncData('titleData', async () => {
+  const runtimeConfig = useRuntimeConfig();
+  const posthog = new PostHog(
+    runtimeConfig.public.posthogPublicKey,
+    { host: runtimeConfig.public.posthogHost }
+  );
+
+  let returnedValue = '';
+  const distinctId = 'placeholder-user-id'
+  const isFlagEnabled = await posthog.isFeatureEnabled('my-cool-flag', distinctId);
+  if (isFlagEnabled) {
+    returnedValue = 'Our flag is enabled!';
+  } else {
+    returnedValue = 'Our flag is disabled!'
+  }
+  return returnedValue;
+});
+const title = computed(() => titleData.value);
+
+</script>
+```
+
+Now when you reload your app, the flicker is gone. 
+
+### Setting the correct `distinctId`
+
+You may have noticed that we set `distinctId = 'placeholder-user-id'` in our flag call above. In production apps, to ensure you fetch the correct flag value for your user, `distinctId` should be set to their unique ID. 
+
+For logged-in users, you typically use their email as their `distinctId`. However, for logged-out users, you can use the `distinct_id` property from their PostHog cookie:
+
+```vue file=app.vue
```

Comment:
Can we handle for the situation when the cookie doesn't exist? Right now, it is just blank. No need to add generating an anonymous ID, just an if statement is fine.

Reply:
Im not sure I follow, there's already an if statement below?

```
if (cookies && cookies.value) {
```

Reply:
Ahh, maybe change   `let returnedValue = '';` to   `let returnedValue = 'No cookie';` ?